src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/ITensor.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Numerics.Tensors
+{
+    public interface ITensor<TSelf, T>
+        : IEnumerable<T>,
+          IEquatable<TSelf>,
+          IEqualityOperators<TSelf, TSelf, bool>
+        where TSelf : ITensor<TSelf, T>
+        where T : IEquatable<T>
```

Comment:
Did we want this constraint, or did we want to rely on `EqualityComparer<T>.Default` instead?

We've not been entirely consistent with how various APIs are defined, but have been removing such constraints more regularly lately since it restricts what generic code can try to call these APIs.

Reply:
I think we want to remove this but I'll need your help to do so because of how much that trickles down. I can't remove the other API's we want to until we remove this.

Reply:
Removed the constraint.